File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/generate_session.kt
// To parse the JSON, install kotlin's serialization plugin and do:
//
// val json            = Json(JsonConfiguration.Stable)
// val generateSession = json.parse(GenerateSession.serializer(), jsonString)

package GenerateSession

import kotlinx.serialization.*
import kotlinx.serialization.json.*
import kotlinx.serialization.descriptors.*
import kotlinx.serialization.encoding.*

@Serializable
data class GenerateSession (
    val data: Data? = null,
    val status: String? = null
)

@Serializable
data class Data (
    @SerialName("access_token")
    val accessToken: String? = null,

    @SerialName("api_key")
    val apiKey: String? = null,

    @SerialName("avatar_url")
    val avatarURL: String? = null,

    val broker: String? = null,
    val email: String? = null,
    val enctoken: String? = null,
    val exchanges: List<String>? = null,

    @SerialName("login_time")
    val loginTime: String? = null,

    val meta: Meta? = null,

    @SerialName("order_types")
    val orderTypes: List<String>? = null,

    val products: List<String>? = null,

    @SerialName("public_token")
    val publicToken: String? = null,

    @SerialName("refresh_token")
    val refreshToken: String? = null,

    val silo: String? = null,

    @SerialName("user_id")
    val userID: String? = null,

    @SerialName("user_name")
    val userName: String? = null,

    @SerialName("user_shortname")
    val userShortname: String? = null,

    @SerialName("user_type")
    val userType: String? = null
)

@Serializable
data class Meta (
    @SerialName("demat_consent")
    val dematConsent: String? = null
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/generate_session.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Debug, Serialize, Deserialize)]
pub struct GenerateSession {
    pub data: Option<Data>,
    pub status: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Data {
    pub access_token: Option<String>,
    pub api_key: Option<String>,
    pub avatar_url: Option<String>,
    pub broker: Option<String>,
    pub email: Option<String>,
    pub enctoken: Option<String>,
    pub exchanges: Option<Vec<String>>,
    pub login_time: Option<String>,
    pub meta: Option<Meta>,
    pub order_types: Option<Vec<String>>,
    pub products: Option<Vec<String>>,
    pub public_token: Option<String>,
    pub refresh_token: Option<String>,
    pub silo: Option<String>,
    pub user_id: Option<String>,
    pub user_name: Option<String>,
    pub user_shortname: Option<String>,
    pub user_type: Option<String>,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Meta {
    pub demat_consent: Option<String>,
}
